### PR TITLE
provider/aws: fix potential aws_route crashes

### LINKED ISSUE #5865: Crash in terraform 0.6.14 on refreshing route state
I attempted to change a CIDR block in a route, and have ended up suck in a state where terraform will crash on every run.  It appears to be crashing while Refreshing state:

```
panic: runtime error: invalid memory address or nil pointer dereference
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: [signal 0xb code=0x1 addr=0x0 pc=0x1e8517]
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws:
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: goroutine 454 [running]:
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: panic(0xdce9a0, 0x821f02110)
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: 	/opt/go/src/runtime/panic.go:464 +0x3e6
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsRouteExists(0x823ab88a0, 0xab8ee0, 0x822daf440, 0x1af5a80, 0x0, 0x0)
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_route.go:297 +0x407
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0x823256b80, 0x823aba9f0, 0xab8ee0, 0x822daf440, 0x821f75ff0, 0x0, 0x0)
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:188 +0x1b7
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0x823272f60, 0x823a5da80, 0x823aba9f0, 0x1af5c01, 0x0, 0x0)
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:187 +0x1da
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/rpc.(*ResourceProviderServer).Refresh(0x82314c200, 0x823aad1b0, 0x823aad5b0, 0x0, 0x0)
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/rpc/resource_provider.go:345 +0x6a
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: reflect.Value.call(0xd0eec0, 0xfe7c18, 0x13, 0x10955f8, 0x4, 0x82341bed8, 0x3, 0x3, 0x0, 0x0, ...)
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: 	/opt/go/src/reflect/value.go:435 +0x120d
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: reflect.Value.Call(0xd0eec0, 0xfe7c18, 0x13, 0x82341bed8, 0x3, 0x3, 0x0, 0x0, 0x0)
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: 	/opt/go/src/reflect/value.go:303 +0xb1
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: net/rpc.(*service).call(0x823257440, 0x823257400, 0x823148d00, 0x822d80200, 0x822db9ce0, 0xaba080, 0x823aad1b0, 0x16, 0xaba0e0, 0x823aad5b0, ...)
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:383 +0x1c2
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: created by net/rpc.(*Server).ServeCodec
2016/03/24 17:02:35 [DEBUG] terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:477 +0x49d
```

I would appreciate any assistance in tracking this down.

Thanks.

### PR DESCRIPTION
**Pull request by phinze**
Fixes #5865

### CODE REVIEW

File: builtin/providers/aws/resource_aws_route.go [hunk 1]
```diff
@@ -320,6 +326,11 @@ func findResourceRoute(conn *ec2.EC2, rtbid string, cidr string) (*ec2.Route, er
 		return nil, err
 	}
 
+	if len(res.RouteTables) < 1 || res.RouteTables[0] == nil {
```
Comment (catsby) [inline builtin/providers/aws/resource_aws_route.go]: The response is called `resp` here, not `res`. I think we should rename it `resp` in the other method, honestly. `resp` is more telling to me than `res`...

**Comment by catsby** [comment]
The patch looks correct, but there's a typo that is causing Travis to fail. After that :+1:

**Comment by phinze** [comment]
Whoops - thanks for the catch / ping. Fixed and we'll see what Travis thinks of it. :grinning:

**Comment by catsby** [comment]
Travis agrees with the changes :+1:

**Pull request merged**